practices/development.md

```diff
@@ -2,105 +2,90 @@
+When you have an implementation that requires feedback, use GitHub's **Request Review** feature to ask other team members to take a look.
+You can either add all team members as a request, or add specific team members if you believe they'll be particularly helpful in a review.
 
-The Activity Backlog is broken down into these columns:
+% TODO: We should define a more structured process for requesting reviews and how it fits in with our merge policies.
 
-- {guilabel}`Needs Discussion/Refinement` Tasks that require some team discussion around _implementation_. If there are higher-level discussions about the deliverable itself, perhaps the deliverable is not yet ready to be worked on.
-- {guilabel}`Ready to Work` Tasks that are ready to be worked on. Roughly speaking, tasks higher on the list are of higher priority.
-- {guilabel}`In progress` A task that a team member is currently working towards. When you move a task to In progress, indicate that you are working on it (either by adding your username to the card, or “assigning yourself” in the issue/PR associated with it.
-- {guilabel}`Needs review` Tasks that have an implementation that require feedback from others. For example, reviews for pull requests.
-- {guilabel}`Done` Tasks that are complete! When you move a task here, make sure to update any relevant deliverables.
+## Encoding longer-term efforts
 
-There is one Activity Backlog defined [at this GitHub Projects Board](https://github.com/orgs/2i2c-org/projects/5?fullscreen=true).
+Longer-term projects are generally more complex and may be made up of many actions and deliverables to accomplish.
+There is no official way to track long-term projects within 2i2c, but there are a few patterns that may be useful to do so.
 
-(coordination:tasks)=
-## Tasks
+% TODO: We should define a more reliable process for tracking long-term projects
 
-Deliverables of sufficient complexity are broken down into _tasks_. Tasks should be completable by a single person in less than one day (generally, completable with a few hours of focused work at most). Make sure to attach a link between your deliverable and its tasks (e.g., by adding links to issues in the deliverable’s "tasks" checklist)
+### Tracking issues
 
-Tasks are generally encoded as checklist items in a deliverable, then added to the Activity Backlog when the deliverable is being worked on.
+The simplest way to track longer-term efforts is with a **Tracking Issue**.
+This is a GitHub Issue whose job is to keep track of many actions and deliverables over time that are needed to close the issue.
+They are generally encoded as **Task Lists** in the issue's top comment.
+Each item in the list tends to be a deliverable, and can be converted into its own GitHub Issue (e.g., to put on the [Activity Board](coordination:activity-board)) as-needed.
 
-Here’s an example of a deliverable with a few tasks:
+(coordination:project-backlog)=
+### Project Backlogs
```

Comment:
How about instead we just have *one* board, where the goal is purely prioritization? It has just high level tasks, and only contains an ordered list of deliverables that can be picked up. Nothing is automatically put into this board - someone must develop an issue to a point where it is ready to be worked on, and explicitly puts it in this board with appropriate priority ordering. We can then use this to pick up tasks from when needed.

Reply:
Mmm - I like the idea of keeping track of prioritizes and high-level deliverables. Though have a few concerns with "one board to rule them all".

1. It doesn't track the overall structure and plan of work in a given project, so will be less useful at understanding the "big picture" of *why* a particular deliverable should have priority. This will also make it harder to know which deliverables go at the top of the list.
2. Many of our projects are totally different from one another, and some team members will have workstreams that don't overlap (e.g. myself and @damianavila are doing a lot of stuff on Jupyter Book projects...would it feel weird to have those deliverables interwoven with things like `pilot-hubs/` upgrades? How would we prioritize one deliverable over another in these cases?
3. I think the "Someone must develop and issue" process is actually a **really important** process, and I'm a bit worried that we are being kind-of hand-wavy about how it happens. For complex issues, once we've gotten to the point of "somebody just needs to implement it", a lot of work has already been done. How do we use these workflows to ensure that this happens?

Reply:
Agree that (3) is super important, and I responded to some of that in https://github.com/2i2c-org/team-compass/pull/188#discussion_r683591706. I think most work is iterative in different kinds of work, and it is difficult to get them to a point where "somebody just needs to implement it" happens. I think doing that kind of work should just be part of the sprint - maybe for one sprint half the work is just getting an issue fleshed out and decisions made (if necessary) with other people.

For (2), I think it's ok to have them mixed in for now. If absolutely necessary, we can filter by labels (or repos). Let's try it out and see if it is a problem?

For (1), IMO we should rely on the content of the issue, since most of it will be pretty unstructured. I don't know if  moving them across columns in a board provides more information than text in an issue. Since almost all our work interacts with folks from outside 2i2c, and they don't really expect to be using our boards, we should just continue using issues until we hit more specific problems there.

Anyway, I think (3) is the most important thing to do. I don't think adding 1 board per project will help with that though - you still have to remember and check which board to look at, and do the work. Keeping it in the issue, and not treating it differently from implementation work alone will help reduce diffusion of responsibility too

Reply:
How do you imagine this being different from the "Development Backlog" that we have already been using? Is the main difference that it is more disciplined and refined than the current one? It feels like the current dev backlog wasn't working for folks, so I'm trying to understand where this differs to make sure we don't just create another board that folks don't find useful.